### PR TITLE
Add cursor and ruler

### PR DESCRIPTION
**Pull request by mikael-johnsson**
Add a custom cursor and horizontal line when in the body with functionality to toggle on and off.
Left to fix is styling the cursor and removing default cursor when clicking links.

merged non conflicting changes pulled from main

Added infrastructure to create hoverStyles feature

### CODE REVIEW

File: ruler.js [hunk 1]
```diff
@@ -1,13 +1,68 @@
 
-/*This script is used to change the cursor to a wait cursor when the ruler tool is selected. 
-Later on we will change the wait symbol to a ruler.*/
+/*This script is used to change the cursor to a ruler when ruler tool is selected*/
 function ruler(element, data) {
         let checkbox = data.ruler
```
Comment (ShaneByrne0963) [inline ruler.js:4]: Several missing semicolons in this file

**Review by ShaneByrne0963 — CHANGES_REQUESTED** [review]
There's a few missing semicolons in ruler.js. Can you fix them before I merge?

**Comment by mikael-johnsson** [comment]
Missing semicolon added

**Review by ShaneByrne0963 — APPROVED** [review]
LGTM

**Pull request merged**